pkg/kubelet/dockershim/docker_service.go

```diff
@@ -54,12 +56,20 @@ const (
 var internalLabelKeys []string = []string{containerTypeLabelKey, sandboxIDLabelKey}
 
 // NOTE: Anything passed to DockerService should be eventually handled in another way when we switch to running the shim as a different process.
-func NewDockerService(client dockertools.DockerInterface, seccompProfileRoot string, podSandboxImage string) DockerLegacyService {
-	return &dockerService{
+func NewDockerService(client dockertools.DockerInterface, seccompProfileRoot string, podSandboxImage string, networkPlugin network.NetworkPlugin, hairpinMode componentconfig.HairpinMode, nonMasqueradeCIDR string, mtu int) (DockerLegacyService, error) {
+	ds := &dockerService{
 		seccompProfileRoot: seccompProfileRoot,
 		client:             dockertools.NewInstrumentedDockerInterface(client),
 		podSandboxImage:    podSandboxImage,
 	}
+
+	// TODO: include probing and selection of network plugin here or in the dockershim server wrapper
+	err := networkPlugin.Init(&DockerNetworkHost{ds: ds}, hairpinMode, nonMasqueradeCIDR, mtu)
```

Comment:
> More realistic example might be hyper only supporting the vm network plugin.

This is true, hyper needs its own network plugin, probably different from docker. So the idea is moving network plugins to shims.